tests/test_dynamics.py

```diff
@@ -0,0 +1,54 @@
+"""Test class which runs a dynamics protocol"""
+from functools import partial
+
+import numpy as np
+from ase.md import VelocityVerlet
+from ase.md.velocitydistribution import MaxwellBoltzmannDistribution
+from ase.optimize import BFGS
+from pytest import fixture
+
+from cascade.dynamics import DynamicsProtocol, Progress
+from cascade.learning.torchani import ANIModelContents, TorchANI
+from cascade.learning.torchani.build import make_aev_computer, make_output_nets
+from cascade.learning.utils import estimate_atomic_energies
+
+
+@fixture()
+def example_model(example_data) -> ANIModelContents:
+    ref_energies = estimate_atomic_energies(example_data)
+    species = list(ref_energies.keys())
+    aev = make_aev_computer(species)
+    nn = make_output_nets(species, aev)
+    return aev, nn, ref_energies
+
+
+def test_dynamics(example_model, example_data):
+    # Prepare the protocol
+    dyn_steps = [
+        (BFGS, 100, {}, {'fmax': 0.001}, partial(MaxwellBoltzmannDistribution, temperature_K=300)),
+        (VelocityVerlet, 100, {'timestep': 0.5}, {}, None)
+    ]
+    dyn = DynamicsProtocol(processes=dyn_steps)
```

Comment:
Considering vocabulary: what if this whole thing is a dynamics job, it consists of tasks, those tasks may be completed in chunks by cascade, and those chunks have timesteps.